tools/driver/cling.cpp

```diff
@@ -106,6 +107,29 @@ int main( int argc, char **argv ) {
     Interp.loadFile(Lib);
 
   cling::UserInterface Ui(Interp);
+
+  // Get the home directory
+  std::string HomeDir;
+#ifdef __unix__
+  HomeDir = getenv("HOME");
+  HomeDir += "/";
+#elif defined(_WIN32)
+  HomeDir = getenv("USERPROFILE");
+  HomeDir += "\\";
+#endif
+
+  // Look for .clinginit in the home directory and if found process it like a cling script
+  std::string CLingInit = HomeDir + ".clinginit";
+  const std::string Filepath = Interp.lookupFileOrLibrary(CLingInit);
+  if (!Filepath.empty()) {
+    std::ifstream File(Filepath);
+    std::string Line;
+    cling::Interpreter::CompilationResult Result;
+    while (std::getline(File, Line)) {
+      Ui.getMetaProcessor()->process(Line, Result, 0);
+    }
```

Comment:
How's that different from `cling::MetaProcessor::readInputFromFile()`?

Reply:
I had used these statements because if you look further down at [line 146](https://github.com/root-project/cling/blob/e23cf8438e590d2310efb49c073b0115497e4b3e/tools/driver/cling.cpp#L146) cling processes the input scripts in a similar way. Here again, using `cling::MetaProcessor::readInputFromFile()` seems to be the better option.